searx/engines/bing.py

```diff
@@ -208,38 +204,21 @@ def response(resp):
                 e.getparent().remove(e)
         content = extract_text(content)
 
-        # get the real URL either using the URL shown to user or following the Bing URL
+        # get the real URL
         if url.startswith('https://www.bing.com/ck/a?'):
-            url_cite = extract_text(eval_xpath(result, './/div[@class="b_attribution"]/cite'))
-            # Bing can shorten the URL either at the end or in the middle of the string
-            if (
-                url_cite
-                and url_cite.startswith('https://')
-                and '…' not in url_cite
-                and '...' not in url_cite
-                and '›' not in url_cite
-            ):
-                # no need for an additional HTTP request
-                url = url_cite
-            else:
-                # resolve the URL with an additional HTTP request
-                url_to_resolve.append(url.replace('&ntb=1', '&ntb=F'))
-                url_to_resolve_index.append(i)
-                url = None  # remove the result if the HTTP Bing redirect raise an exception
+            # get the first value of u parameter
+            url_query = urlparse(url).query
+            parsed_url_query = parse_qs(url_query)
+            param_u = parsed_url_query["u"][0]
+            # remove "a1" in front
+            encoded_url = param_u[2:]
+            # add padding
+            encoded_url = encoded_url + '=' * (-len(encoded_url) % 4)
+            # decode base64 encoded URL
+            url = base64.b64decode(encoded_url, validate=True).decode()
```

Comment:
With search term `:en !bing foo` I got an exception ..

```
  File "/share/searxng/searx/engines/bing.py", line 218, in response
    url = base64.b64decode(encoded_url, validate=True).decode()
  File "/usr/lib/python3.10/base64.py", line 86, in b64decode
    raise binascii.Error('Non-base64 digit found')
binascii.Error: Non-base64 digit found
```

Reply:
Fixed at least for the query you mentioned